torchdata/datapipes/iter/transform/callable.py

```diff
@@ -414,3 +418,138 @@ def __len__(self) -> int:
         if isinstance(self.datapipe, Sized):
             return len(self.datapipe)
         raise TypeError(f"{type(self).__name__} instance doesn't have valid length")
+
+
+class _BatchAsyncMapperIterDataPipe(IterDataPipe):
+    datapipe: IterDataPipe
+    async_fn: Callable
+
+    def __init__(
+        self,
+        source_datapipe: IterDataPipe,
+        async_fn: Callable,
+        input_col=None,
```

Comment:
curious if it will becomes something like `input_cols/output_cols` or it always has to be single column?

Reply:
Current state:
- `input_col` accepts multiple elements
- `output_col` is not supported